airflow/api_fastapi/execution_api/datamodels/asset_event.py

```diff
@@ -0,0 +1,66 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from datetime import datetime
+
+from pydantic import Field, field_validator
+
+from airflow.api_fastapi.core_api.base import BaseModel, StrictBaseModel
+from airflow.sdk.execution_time.secrets_masker import redact
+
+
+class DagRunAssetReference(StrictBaseModel):
+    """DAGRun serializer for asset responses."""
+
+    run_id: str
+    dag_id: str
+    logical_date: datetime | None
+    start_date: datetime
+    end_date: datetime | None
+    state: str
+    data_interval_start: datetime | None
+    data_interval_end: datetime | None
+
+
+class AssetEventResponse(BaseModel):
+    """Asset event schema with fields that are needed for Runtime."""
+
+    id: int
+    asset_id: int
+    uri: str | None = Field(alias="uri", default=None)
+    name: str | None = Field(alias="name", default=None)
+    group: str | None = Field(alias="group", default=None)
```

Comment:
Similarly, these should just be one asset object? This is a bit more important than source_task_instance since the `extra` fields on Asset and AssetEvent conflit in name.

Reply:
yep, sounds like a good idea. I'll make this change in this PR

Reply:
btw should we make models.AssetEvent as models.AssetEventModel (probably in another PR?)

Reply:
I think we should not add new types (AssetEvent and DagRunAssetReference) to definitions. Things in definitions are intended to be used by users, but we don’t want users to instantiate either of them. (Are they actually used anywhere?) We can expose the Pydantic models instead, either directly or with some processing.

Reply:
got it. I thought we want these exposed object in definitions, but yep, makes sense. Will update it

Reply:
It's now removed

Reply:
I guess I'll just expose `AssetEventsResult`. Should we also remove `AssetAliasEvent` from definition (probably in next PR though)?

Reply:
Make it return `AssetEventsResult`